### DISCUSSION TITLE
REDIS_CONNECTION example works, unless you don't have redis server running

### DISCUSSION THREAD

**Discussion by str11ngfello** [TypedProject/ts-express-decorators#2522] [General]
I've been using tsed.io's ioredis intergration for almost a year. The docs say to do this:

![image](https://github.com/tsedio/tsed/assets/877000/0929f84a-64ac-4879-ac4c-89583437ad21)

And then you can inject in your other code like this:

![image](https://github.com/tsedio/tsed/assets/877000/61ef069b-bd38-4a71-b8a8-e77037a49bac)

As long as you have a redis server running of course.

However, recently I created a tsed cli command to generate types (found that in another dicussion here) and that command passes in a blank object for redis like so:

![image](https://github.com/tsedio/tsed/assets/877000/b67c31ff-887c-4aba-8011-56187f438e6d)

All this was fine, as long as redis is running. But the cli command just generates types so when I run it in an environment without redis the command repeated trys to connect to redis :D

![image](https://github.com/tsedio/tsed/assets/877000/261b7ed0-ea33-4860-a35c-7af6d15d63bc)

I **Think** it's because the way the sample code works for REDIS_CONNECTION provide. It looks like the ioredis Redis object will start a connection just by instantiating it. 

How can I modify the sample REDIS_CONNECTION provider to be smart enough to know that I firing up a command and it should not let iosredis try to connect. I saw ioredis has a "lazyConnect" option but am unsure how to use that with this code:

![image](https://github.com/tsedio/tsed/assets/877000/aa543ecf-7023-4de0-9f7d-77f11c44df99)

I hope I'm close but just missing something. Anyone have any ideas?

**Comment by str11ngfello** [comment]
btw, the cli command doesn't finish and repeatedly waits for redis to connect. This is why it's a blocker. If it weren't a hang it might be okay with me. But it's blocking the command from running. Of course I can fire up a local redis docker container and get by, but my CI environment won't have a redis server to connect so it'll fail in that case.

**Reply by Romakita** [reply]
Hello @str11ngfello 

It's a bit strange because when the ioredis options is `[]`, nothing should be initialized. And I confirm that works, because I already had this issue on my own project (Generate types and custom commands). 

I also a more complex scenario, because, some custom commands needs redis connection and for the command to generates types isn't necessary.

Here is my code for the CLI:

```ts
CliCore.bootstrap({
  ...configuration,
  mongoose: mongooseConfig,
  cache: cacheConfig,
  ioredis: redisConfig, // I let DB configuration because sometime I need it.
  issuers,
  server: Server,
  logstash: false,
  httpClient: {
    transformOperationId(operationId: string, routeNameInfo: any, raw: any) {
      return raw.moduleName === "oidc" ? routeNameInfo.original.replace("oidc", "") : operationId;
    }
  },
  // add your custom commands here
  commands: [GenerateHttpClientCmd, DeprecatedRoutesCmd, ProductCalendarsCmd, StatusPageCmd],
  configResolvers: [databaseResolver] // I provide a configuration Resolver to manage when I need db or not.
})
```

Here is the databaseResolver:

```typescript
export function databaseResolver(settings: Record<string, any>) {
  if (!get(settings, "command.enableFeatures").includes("database")) {
    settings.mongoose = [];
    settings.ioredis = [];
    settings.cache = false;
  }
}
```
Using this snippet, I can now add a metadata on `Command` decorator to know if the command needs some extra features to works:

```ts
@Command({
  name: "calendar",
  description: "Generate the calendars data",
  options: {
   
  },
  enableFeatures: ["database"],
  bindLogger: false
})
export class ProductCalendarsCmd implements CommandProvider { }
```

You can try that, but for me your actual code should works as expected (while ioredis option is empty). 

You can take a look on the code here:
https://github.com/tsedio/tsed/blob/production/packages/orm/ioredis/src/utils/registerConnectionProvider.ts


If your goal is to generate source code from the swagger.json you already have a cli module for that: "@tsed/cli-generate-http-client":
https://github.com/tsedio/tsed-cli/blob/master/packages/cli-generate-http-client/src/commands/GenerateHttpClientCmd.ts

The code example in the discussion is probably obsolete ;)

See you
Romain
